### PR TITLE
JAMES-3589 Backport MailetContainerImpl with adequate fix

### PR DESCRIPTION
**Pull request by chibenwa**
See https://github.com/apache/james-project/pull/448

Requires https://github.com/apache/james-project/pull/451 to be :green_apple:

### CODE REVIEW

**Comment by Arsnael** [comment]
Some bindings issues it seems 

```
[ERROR] shouldUseMemoryMailboxManager{GuiceJamesServer}  Time elapsed: 0.071 s  <<< ERROR!
com.google.inject.CreationException: 
Unable to create injector, see the following errors:

1) A binding to org.apache.james.mailetcontainer.api.MailProcessor was already configured at org.apache.james.mailetcontainer.impl.MailetContainerImplModule.configure(MailetContainerImplModule.java:75) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailetcontainer.impl.CustomMailetProcessingModule -> org.apache.james.mailetcontainer.impl.MailetContainerImplModule).
  at org.apache.james.modules.server.CamelMailetContainerModule.configure(CamelMailetContainerModule.java:95) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.modules.MailetProcessingModule -> org.apache.james.modules.server.CamelMailetContainerModule)

2) A binding to org.apache.james.mailetcontainer.impl.JamesMailSpooler$Configuration was already configured at org.apache.james.mailetcontainer.impl.MailetContainerImplModule.spoolerConfiguration() (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailetcontainer.impl.CustomMailetProcessingModule -> org.apache.james.mailetcontainer.impl.MailetContainerImplModule).
  at org.apache.james.modules.server.CamelMailetContainerModule.spoolerConfiguration(CamelMailetContainerModule.java:130) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.modules.MailetProcessingModule -> org.apache.james.modules.server.CamelMailetContainerModule)

3) A binding to org.apache.james.mailrepository.memory.MailRepositoryStoreConfiguration was already configured at org.apache.james.modules.server.MailStoreRepositoryModule.provideConfiguration() (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailetcontainer.impl.CustomMailetProcessingModule -> org.apache.james.modules.server.MailStoreRepositoryModule).
  at org.apache.james.modules.server.MailStoreRepositoryModule.provideConfiguration(MailStoreRepositoryModule.java:66) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.modules.MailetProcessingModule -> org.apache.james.modules.server.MailStoreRepositoryModule)

3 errors
```

**Comment by Arsnael** [comment]
Read it by the way

File: tmail-backend/apps/memory/src/main/java/com/linagora/tmail/james/app/MemoryServer.java [hunk 1]
```diff
@@ -38,6 +46,18 @@
 import com.linagora.tmail.james.jmap.ticket.TicketRoutesModule;
 
 public class MemoryServer {
+    public static final Module IN_MEMORY_SERVER_MODULE = Modules.combine(
+        new CustomMailetProcessingModule(),
+        new BlobMemoryModule(),
+        new DeletedMessageVaultModule(),
+        new BlobExportMechanismModule(),
```
Comment (Arsnael) [inline tmail-backend/apps/memory/src/main/java/com/linagora/tmail/james/app/MemoryServer.java:53]: Is that one necessary? (and the DeletedMessageVaultModule above too?)

Comment (chibenwa) [reply tmail-backend/apps/memory/src/main/java/com/linagora/tmail/james/app/MemoryServer.java:53]: Clean up should come in a follow up PR (they were here before, their removal is an enhancement)

Technically speaking DTM is a feature of OpenPaaS thus its support is compulsory in TMail.

**Review by Arsnael — APPROVED** [review]

**Pull request merged**